### PR TITLE
Add Hybrid planning action definitions

### PR DESCRIPTION
**Pull request by sjahr**
This PR adds the action interfaces for MoveIt 2's hybrid planning feature (more information about it can be found in these moveit2 issues: [#300](https://github.com/ros-planning/moveit2/issues/300) & [#433](https://github.com/ros-planning/moveit2/issues/433))

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -108,6 +108,9 @@ set(action_files
   "action/MoveGroupSequence.action"
   "action/Pickup.action"
   "action/Place.action"
+  "action/LocalPlanner.action"
+  "action/GlobalPlanner.action"
+  "action/HybridPlanning.action"
```
Comment (AdamPettinger) [inline CMakeLists.txt:113]: Only a tiny comment, this is `*Planner`, `*Planner`, `*Planning`, maybe the last should be `action/HybridPlanner`?

**Review by AdamPettinger — APPROVED** [review]
These actions make sense for the hybrid planner, and this PR should probably be merged before https://github.com/ros-planning/moveit2/pull/488

Just left one small nitpicking comment to consider making the action naming consistent

**Review by henningkayser — COMMENTED** [review]
I agree with @AdamPettinger, the names could be a little bit more consistent. I'm wondering if there is a better name than `action/HybridPlanning` which indicates that the motion is executed at the same time...

File: action/LocalPlanner.action [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# Additional local constraints for the local planner (Global trajectory is received directlyfrom the global planning component)
+
+Constraints[] local_constraints
+# Optional commands for the local planning component i.e. abort current planning operation
```
[the remark this answers is not in the document]
Comment (sjahr) [reply action/LocalPlanner.action:4]: Yes, you are right. At this point, the command string is not necessary and the example I've given was not correct. When/ If the architecture's implementation is extended to handle concurrent and overlapping goal requests for both, the local and the global planner, this field might become useful to determine the relationship between multiple goals (i.e. preemptive or sequential). But for now, I'll remove it

**Comment by sjahr** [comment]
The latest commit  breaks the API in favor of the new plugin interface for the global planner ([moveit2 #585](https://github.com/ros-planning/moveit2/pull/585))

File: action/GlobalPlanner.action [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# Motion planning request to the global planner
+string planning_group
+moveit_msgs/MotionSequenceRequest desired_motion_sequence
```
Comment (henningkayser) [inline action/GlobalPlanner.action:3]: I'm not really convinced of this name. Simply `request` would be fine imo

Comment (tylerjw) [reply action/GlobalPlanner.action:3]: I think calling this motion_sequence (As it is in the HybridPlanner.action) is actually clearer when used in code as it is a sub-message of the action goal.

**Comment by tylerjw** [comment]
Closing in favor of #135 where I will continue this work.

File: action/GlobalPlanner.action [hunk 4]
```diff
@@ -0,0 +1,11 @@
+# Motion planning request to the global planner
+moveit_msgs/MotionPlanRequest request
+
+# Optional commands for the global planning component i.e. abort current planning operation
```
[the remark this answers is not in the document]
Comment (AndyZe) [reply action/GlobalPlanner.action]: This field was removed, similar to LocalPlanner.action

File: action/HybridPlanning.action [hunk 5]
```diff
@@ -0,0 +1,12 @@
+# Motion planning request to the hybrid planning architecture
+string planning_group
+moveit_msgs/MotionSequenceRequest motion_sequence
+---
+# Result of the hybrid planning
+moveit_msgs/MoveItErrorCodes error_code
+
+# Error message with more detailed information about the hybrid planning result
+string error_message
```
[the remark this answers is not in the document]
Comment (AndyZe) [reply action/HybridPlanning.action:9]: sounds very nitpicky

**Pull request closed without merging**